src/Tools/dotnet-monitor/CollectionRules/CollectionRuleService.cs

```diff
@@ -32,16 +36,27 @@ CollectionRulesConfigurationProvider provider
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _provider = provider ?? throw new ArgumentNullException(nameof(provider));
             _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+
+            BoundedChannelOptions containersToRemoveChannelOptions = new(1000)
+            {
+                SingleReader = true,
+                SingleWriter = true
+            };
+
+            Channel<CollectionRuleContainer> containersToRemoveChannel =
+                Channel.CreateBounded<CollectionRuleContainer>(containersToRemoveChannelOptions);
+            _containersToRemoveReader = containersToRemoveChannel.Reader;
+            _containersToRemoveWriter = containersToRemoveChannel.Writer;
```

Comment:
nit: Consider completing the writer during disposal

Reply:
Thought I did that but maybe I didn't push it. Thanks for catching that.